src/plugins/es_ui_shared/static/forms/hook_form_lib/components/form_data_provider.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useState, useEffect, useRef } from 'react';
+
+import { Form, FormData } from '../types';
+import { Subscription } from '../lib';
+
+interface Props {
+  children: (formData: FormData) => JSX.Element;
+  form: Form<FormData>;
+  pathsToWatch?: string | string[];
+}
+
+export const FormDataProvider = ({ children, form, pathsToWatch }: Props) => {
+  const [formData, setFormData] = useState<FormData>({});
+  const previousState = useRef<FormData>({});
+  const subscription = useRef<Subscription | null>(null);
+
+  useEffect(() => {
+    let didUnsubscribe = false;
```

Comment:
Did you run into a situation where this `didUnsubscribe` was needed to block the listener from running? I'm just wondering because I'm not sure that could ever happen?

Reply:
I am going to double-check but this might be a leftover of a React error I was having about trying to update the state on an unmounted component. I made a few changes and maybe this is not necessary anymore. Thanks for pointing it out!

Reply:
It's was not needed anymore and has been removed 👍